File: demo/src/main/java/com/bytedance/scenedemo/navigation/popto/PopToScene_1.java
package com.bytedance.scenedemo.navigation.popto;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;

import com.bytedance.scene.Scene;
import com.bytedance.scenedemo.utility.ColorUtil;

/**
 * Created by JiangQi on 8/2/18.
 */
public class PopToScene_1 extends Scene {
    @NonNull
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        LinearLayout layout = new LinearLayout(getActivity());
        layout.setOrientation(LinearLayout.VERTICAL);

        final Bundle bundle = getArguments();
        int index = 0;
        if (bundle != null) {
            index = bundle.getInt("index");
        }
        layout.setBackgroundColor(ColorUtil.getMaterialColor(getResources(), index));

        Button button = new Button(getActivity());
        button.setText("现在是PopToScene1，点击跳到 PopToScene_2");
        layout.addView(button);
        final int finalIndex = index;
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();
                bundle.putInt("index", finalIndex + 1);

                getNavigationScene().push(PopToScene_2.class, bundle);
            }
        });

        return layout;
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/navigation/popto/PopToScene_1.kt
package com.bytedance.scenedemo.navigation.popto

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import com.bytedance.scene.Scene
import com.bytedance.scene.ktx.requireNavigationScene
import com.bytedance.scenedemo.R
import com.bytedance.scenedemo.utility.ColorUtil

/**
 * Created by JiangQi on 8/2/18.
 */
class PopToScene_1 : Scene() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        return inflater.inflate(R.layout.basic_layout, container, false) as ViewGroup
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view.setBackgroundColor(ColorUtil.getMaterialColor(resources, 2))
        val name = view.findViewById<TextView>(R.id.name)
        name.text = requireNavigationScene().stackHistory
        val btn = view.findViewById<Button>(R.id.btn)
        btn.text = getString(R.string.nav_pop_to_btn_2)
        btn.setOnClickListener { requireNavigationScene().push(PopToScene_2::class.java) }
    }
}